src/resolve/resolved-dns-cache.c

```diff
@@ -442,8 +442,9 @@ static int dns_cache_put_positive(
          * RRs are still valid, hence find the lowest here */
         min_ttl = MIN(dns_answer_min_ttl(answer), rr->ttl);
 
-        /* New TTL is 0? Delete this specific entry... */
-        if (min_ttl <= 0) {
+        /* New TTL is 0? Delete this specific entry...
+         * In RFC6762 10.1 for Good Bye packet TTL is 1 */
+        if (min_ttl <= 1) {
```

Comment:
hmm, as I read the RFC we should instead when min_ttl is zero go through all our matching cached entries and set the stored TTL to 1s.

Reply:
How about the following? When `min_ttl == 1`, find matching existing positive cache entries and update its TTL and cache lifetime `DnsCacheItem.until`.

Reply:
It makes sense to update the ttl/until of existing matching cache entries to 1 second when we get a TTL==0.
However, there also needs to be some sort of callback after one second to check if the entry was updated in the meantime, and remove the entry if no update was received.
Without this, the entry lingers in the cache indefinitely until the cache is pruned the next time.
We stumbled into this behavior while implementing #22532, with the aim of notifying clients when a service expires.